monai/engines/evaluator.py

```diff
@@ -259,8 +266,12 @@ def __init__(
             to_kwargs=to_kwargs,
             amp_kwargs=amp_kwargs,
         )
-
-        self.network = network
+        if compile:
+            assert pytorch_after(2, 1)
+            self.network = torch.compile(network, **compile_kwargs)
+        else:
+            self.network = network
```

Comment:
```suggestion
        if compile:
            if pytorch_after(2, 1):
                self.network = torch.compile(network, **compile_kwargs)
            else:
                warnings.warn("Network compilation (compile=True) not supported for Pytorch versions before 2.2, no compilation done")
```

I'd suggest warning if version is too old rather than getting an assert fail.